### PR TITLE
EI-1075 996 extend phase 1 eligibility period

### PR DESCRIPTION
**Pull request by ruslanss**

### CODE REVIEW

File: src/SFA.DAS.EmployerIncentives.Domain/Interfaces/IIncentivePaymentProfilesService.cs [hunk 1]
```diff
@@ -6,6 +6,6 @@ namespace SFA.DAS.EmployerIncentives.Domain.Interfaces
 {
     public interface IIncentivePaymentProfilesService
     {
-        Task<IEnumerable<IncentivePaymentProfile>> Get();
+        Task<IncentivesConfiguration> Get();
```
Comment (Stephek) [inline src/SFA.DAS.EmployerIncentives.Domain/Interfaces/IIncentivePaymentProfilesService.cs:9]: This interface doesn't need to change.  The parent of the data is still an IncentivePaymentProfile object.  Extra data has been added to it and some moved around.

File: src/SFA.DAS.EmployerIncentives.Domain/ValueObjects/IncentivesConfiguration.cs [hunk 2]
```diff
@@ -0,0 +1,65 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using SFA.DAS.EmployerIncentives.Abstractions.Domain;
+using SFA.DAS.EmployerIncentives.Domain.ApprenticeshipIncentives.Exceptions;
+using SFA.DAS.EmployerIncentives.Enums;
+
+namespace SFA.DAS.EmployerIncentives.Domain.ValueObjects
+{
+    public class IncentivesConfiguration
```
Comment (Stephek) [inline src/SFA.DAS.EmployerIncentives.Domain/ValueObjects/IncentivesConfiguration.cs:10]: This is not required and is not a business object.  There is already an IncentivePaymentProfile that does this.

Configuration is an infrastructure concern.

File: src/SFA.DAS.EmployerIncentives.Enums/IncentivePhase.cs [hunk 3]
```diff
@@ -0,0 +1,9 @@
+﻿namespace SFA.DAS.EmployerIncentives.Enums
+{
+    public enum IncentivePhase : byte
```
Comment (Stephek) [inline src/SFA.DAS.EmployerIncentives.Enums/IncentivePhase.cs:3]: This should be a value object Phase with a constructor that accepts the phase string or byte (or whatever is to be stored in the db against the application or incentive)

File: src/SFA.DAS.EmployerIncentives.Queries/NewApprenticeIncentive/GetApplication/GetApplicationQueryHandler.cs [hunk 4]
```diff
@@ -36,7 +36,7 @@ public async Task<GetApplicationResponse> Handle(GetApplicationRequest query, Ca
             return response;
         }
 
-        private bool IsNewAgreementRequired(IEnumerable<IncentiveApplicationApprenticeshipDto> applicationApprenticeships,  IEnumerable<IncentivePaymentProfile> paymentProfiles, int signedAgreementVersion)
+        private bool IsNewAgreementRequired(IEnumerable<IncentiveApplicationApprenticeshipDto> applicationApprenticeships,  IncentivesConfiguration paymentProfiles, int signedAgreementVersion)
```
Comment (Stephek) [inline src/SFA.DAS.EmployerIncentives.Queries/NewApprenticeIncentive/GetApplication/GetApplicationQueryHandler.cs:39]: No need to change this interface.

File: src/SFA.DAS.EmployerIncentives.Domain/Interfaces/IIncentivePaymentProfilesService.cs (hunk 1 above)
Comment (Stephek) [inline src/SFA.DAS.EmployerIncentives.Domain/Interfaces/IIncentivePaymentProfilesService.cs:9]: Add a new Get overload  Task<IncentivePaymentProfile> Get(Phase phase) so the details can be loaded for the application/incentive if required.

**Review by Stephek — CHANGES_REQUESTED** [review]

File: src/SFA.DAS.EmployerIncentives.Domain/ValueObjects/IncentivesConfiguration.cs (hunk 2 above)
Comment (ruslanss) [reply src/SFA.DAS.EmployerIncentives.Domain/ValueObjects/IncentivesConfiguration.cs]: renamed

File: src/SFA.DAS.EmployerIncentives.Domain/Interfaces/IIncentivePaymentProfilesService.cs (hunk 1 above)
Comment (ruslanss) [reply src/SFA.DAS.EmployerIncentives.Domain/Interfaces/IIncentivePaymentProfilesService.cs]: we did not intend to store Phase Id in this user story

File: src/SFA.DAS.EmployerIncentives.Enums/IncentivePhase.cs (hunk 3 above)
Comment (ruslanss) [reply src/SFA.DAS.EmployerIncentives.Enums/IncentivePhase.cs:3]: why?

**Review by Stephek — APPROVED** [review]

**Review by David-B-Read — APPROVED** [review]

**Pull request closed without merging**